Modules/galaxy/galaxy.py

```diff
@@ -0,0 +1,276 @@
+        """
+        Finds the furthest star from away from "start" and towards "end",
+        with a limitation of no more than "distance" light-years.
+
+        Args:
+            start (StarSystem): The system to begin our journey in.
+            end (StarSystem): The system which we want to head towards.
+            distance (int): The maximum distance we can travel towards the end system.
+
+        Returns:
+            A ``StarSystem`` object representing the nearest star found after travelling
+            ``distance`` light years from the start system towards the end system.
+
+            Returns None if no system was found after travelling.
+        """
+
+        # Calculate the direction of travel required to go from ``start`` to ``end``.
+        normal = (end.position - start.position).normal()
+        # Travel in that direction for ``distance`` light years.
+        new_position = start.position + (normal * distance)
+        nearest = await self._find_nearest_systems(new_position.x,
+                                                   new_position.y,
+                                                   new_position.z)
+        # Check to ensure the data set is not missing or empty.
+        if nearest:
+            return await self.find_system_by_name(nearest[0])
+
+    async def _find_nearest_systems(self,
+                                    x: float,
+                                    y: float,
+                                    z: float,
+                                    limit: int = 10) -> Optional[List[str]]:
+        """
+        Given a set of galactic coordinates, find the closest star systems.
+
+        Args:
+
+            x (float): The galactic X coordinate to center the search on.
+            y (float): The galactic Y coordinate to center the search on.
+            z (float): The galactic Z coordinate to center the search on.
+            limit (int): The maximum number of nearest systems to return.
+
+        Returns:
+            A ``list`` of ``StarSystem`` objects representing the nearest star systems, up to
+            ``limit`` entries, or ``None`` if no nearby sysems could be found.
+        """
+
+        nearest = await self._call("nearest",
+                                   {"x": x,
+                                    "y": y,
+                                    "z": z,
+                                    "aggressive": "1",
+                                    "limit": limit})
+        if nearest['data']:
+            return [neighbor['name'] for neighbor in nearest['data']]
+
+    @staticmethod
+    def _match_main_star(system_name: str, body_name: str) -> bool:
+        """
+        Attempts to match a star system's name with the actual name of its main star,
+        case insensitively.
+
+        Elite: Dangerous is a bit inconsistent with naming conventions, so the main star
+        can either share the same name as the system, or have an "A" appended to the end.
```

Comment:
Elite only adds a `A` to the end of the star's name if and only if there is more than one star in the system.
```suggestion
        can either share the same name as the system, or have an "A" appended to the end.
        Elite adds a `A` to the end if the system contains more than one star, 
        this routine will check for this automatically
```